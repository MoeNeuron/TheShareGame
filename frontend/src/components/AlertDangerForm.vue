<template>
  <b-row class="justify-content-md-center">
    <b-alert show variant="danger"
             v-if="isValid !== null && !isValid">{{ errMessage }}
    </b-alert>
  </b-row>
</template>

<script>

import { BRow, BAlert } from 'bootstrap-vue';

export default {
  name: 'AlertDangerForm',
  components: {
    BRow, BAlert,
  },
  props: ['isValid', 'errMessage'],
};
</script>

<style scoped>

</style>
