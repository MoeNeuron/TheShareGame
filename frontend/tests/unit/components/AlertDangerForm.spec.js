import { createLocalVue, mount } from '@vue/test-utils';

import AlertDangerForm from '@/components/AlertDangerForm.vue';

const localVue = createLocalVue();

describe('AlertDangerForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(AlertDangerForm, {
      localVue,
      propsData: {
        isValid: false,
        errMessage: 'Error 404!',
      },
    });
  });

  it('renders ErrMessage', () => {
    expect(wrapper.html()).toContain('Error 404!');

    wrapper = mount(AlertDangerForm, {
      localVue,
      propsData: {
        isValid: true,
        errMessage: 'Error 404!',
      },
    });

    expect(wrapper.html()).not.toContain('Error 404!');
  });

  it('matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
