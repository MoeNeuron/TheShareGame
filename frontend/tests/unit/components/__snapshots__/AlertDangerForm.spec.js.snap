// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDangerForm matches Snapshot 1`] = `
<div class="row justify-content-md-center">
  <div role="alert" aria-live="polite" aria-atomic="true" class="alert alert-danger">
    <!---->Error 404!
  </div>
</div>
`;
